### PR TITLE
Add method that show base installation info about gensim & related packages. Fix #1902

### LINKED ISSUE #1902: Add method for showing base info about gensim
Intro
----
When users report something, we need to know more information (not only gensim version, versions of related packages, OS and so on), and user need to copy-paste&run something like this

````
Please run the following snippet and paste the output below.
import platform; print(platform.platform())
import sys; print("Python", sys.version)
import numpy; print("NumPy", numpy.__version__)
import scipy; print("SciPy", scipy.__version__)
import gensim; print("gensim", gensim.__version__)
from gensim.models import word2vec;print("FAST_VERSION", word2vec.FAST_VERSION)
````

We can easily add it as method to gensim and use as `python -m gensim.scripts.package_info --info`

Todo
-------
Implement this method, information, that should be showed
- gensim version
- python version
- numpy version
- scipy version
- os version
- smart_open version
- `FAST_VERSION` variable
- path (where gensim installed)
- ??? (maybe something else)

### PR DESCRIPTION
**Pull request by sharanry**
Fixes #1902

### CODE REVIEW

File: gensim/__main__.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import gensim
+import argparse
+
+if __name__ == "__main__":
+
+        parser = argparse.ArgumentParser()
+        parser.add_argument("--info", action="store_true", help="Display version of Gensim and its dpendencies")
+        opt = parser.parse_args()
+        if opt.info:
+                print(platform.platform())
```
Comment (menshikh-iv) [inline gensim/__main__.py]: Please move it do distinct function (that returns `dict` with information)
```python

from gesnim import package_info

info = package_info()
```

and print it here in nice manner

File: gensim/__main__.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import platform
+import sys
+import numpy
+import scipy
+import gensim
+import argparse
+
+if __name__ == "__main__":
+
+        parser = argparse.ArgumentParser()
```
Comment (menshikh-iv) [inline gensim/__main__.py]: Where is the description? Should use `__doc__` variable like here https://github.com/RaRe-Technologies/gensim/commit/117d447087b2e9a36fda1279d837e045fefd1558

File: gensim/__main__.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import platform
+import sys
+import numpy
+import scipy
+import gensim
+import argparse
+
```
Comment (menshikh-iv) [inline gensim/__main__.py]: Missing docstring for the module, should looks like https://github.com/RaRe-Technologies/gensim/commit/117d447087b2e9a36fda1279d837e045fefd1558#diff-cbaf05cc913e951e3205d150e33edf2eR9

Comment (menshikh-iv) [reply gensim/__main__.py]: In addition, don't forget about `.. program-output:: ` section

File: gensim/__main__.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+if __name__ == "__main__":
+
```
Comment (menshikh-iv) [inline gensim/__main__.py]: remove this line

File: gensim/__main__.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+        parser = argparse.ArgumentParser()
+        parser.add_argument("--info", action="store_true", help="Display version of Gensim and its dpendencies")
```
Comment (menshikh-iv) [inline gensim/__main__.py]: indentation should be 4 spaces (not 8)

File: gensim/__main__.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+        opt = parser.parse_args()
+        if opt.info:
+                print(platform.platform())
+                print("Python", sys.version)
+                print("NumPy", numpy.__version__)
+                print("SciPy", scipy.__version__)
+                print("gensim", gensim.__version__)
```
Comment (menshikh-iv) [inline gensim/__main__.py]: missing `smart_open`, `FAST_VERSION` and path where gensim installed

File: gensim/__main__.py (hunk 3 above)
Comment (menshikh-iv) [reply gensim/__main__.py]: Also, need to add `.rst` to `gensim/docs/src`

Comment (sharanry) [reply gensim/__main__.py]: On it

File: gensim/__main__.py (hunk 6 above)
Comment (sharanry) [reply gensim/__main__.py]: for smart_open `smart_open.__version__` doesn't work as the variable does not exist.

Comment (menshikh-iv) [reply gensim/__main__.py]: Try to found other way

File: gensim/__main__.py (hunk 3 above)
Comment (sharanry) [reply gensim/__main__.py]: > Also, need to add .rst to gensim/docs/src

How do I do this?

Comment (menshikh-iv) [reply gensim/__main__.py]: Look into `gensim/docs/src/scripts` for example, you'll see several `*.rst` files, you need to add something similar to needed place (same as in package hierarchy).

**Comment by menshikh-iv** [comment]
Thanks @sharanry :+1:

**Pull request merged**